app/javascript/packages/document-capture/components/in-person-location-full-address-entry-post-office-search-step.spec.tsx

```diff
@@ -154,6 +154,124 @@ describe('InPersonLocationFullAddressEntryPostOfficeSearchStep', () => {
     expect(errors).to.have.lengthOf(4);
   });
 
+  it('displays a validation error for an invalid ZIP code length (length = 1)', async () => {
+    const { findByText, findByLabelText, findAllByText } = render(
+      <InPersonContext.Provider
+        value={{
+          inPersonOutageMessageEnabled: false,
+          inPersonOutageExpectedUpdateDate: 'January 1, 2024',
+          inPersonFullAddressEntryEnabled: true,
+          usStatesTerritories: [['Delaware', 'DE']],
+        }}
+      >
+        <InPersonLocationFullAddressEntryPostOfficeSearchStep {...DEFAULT_PROPS} />,
+      </InPersonContext.Provider>,
+      { wrapper },
+    );
+
+    await userEvent.type(
+      await findByLabelText('in_person_proofing.body.location.po_search.zipcode_label'),
+      '9',
+    );
+
+    await userEvent.click(
+      await findByText('in_person_proofing.body.location.po_search.search_button'),
+    );
+
+    const errors = await findAllByText('idv.errors.pattern_mismatch.zipcode');
+    expect(errors).to.have.lengthOf(1); 
+  });
+
+  it('displays a validation error for an invalid ZIP code length (length = 8)', async () => {
+    const { findByText, findByLabelText, findAllByText } = render(
+      <InPersonContext.Provider
+        value={{
+          inPersonOutageMessageEnabled: false,
+          inPersonOutageExpectedUpdateDate: 'January 1, 2024',
+          inPersonFullAddressEntryEnabled: true,
+          usStatesTerritories: [['Delaware', 'DE']],
+        }}
+      >
+        <InPersonLocationFullAddressEntryPostOfficeSearchStep {...DEFAULT_PROPS} />,
+      </InPersonContext.Provider>,
+      { wrapper },
+    );
+
+    await userEvent.type(
+      await findByLabelText('in_person_proofing.body.location.po_search.zipcode_label'),
+      '99999-892',
+    );
+
+    await userEvent.click(
+      await findByText('in_person_proofing.body.location.po_search.search_button'),
+    );
+
+    const errors = await findAllByText('idv.errors.pattern_mismatch.zipcode');
+    expect(errors).to.have.lengthOf(1); 
+  });
+
+  it('does not display a validation error for a valid ZIP code length (length = 5)', async () => {
```

Comment:
This test and following are expensive! 1029ms & 1046ms. Other tests in this file find return post offices so kind of indirectly prove no validation error (else user would not have been able to continue.) I am looking for an alternative solution but considering just deleting test does not display a validation error for a valid ZIP code length (length = 5) and does not display a validation error for a valid ZIP code length (length = 9). Thoughts?